frontend/src/modules/machinery/components/Machinery.js

```diff
@@ -72,12 +77,23 @@ export default class Machinery extends React.Component<Props> {
             return null;
         }
 
+        const showResetButton = !machinery.entity && machinery.sourceString;
+
         return (
             <section className='machinery'>
                 <div className='search-wrapper clearfix'>
-                    <label htmlFor='machinery-search'>
-                        <div className='fa fa-search'></div>
-                    </label>
+                    {showResetButton ? (
+                        <label htmlFor='machinery-search'>
```

Comment:
Looks like the `<label>` element is the same, so you can move the condition inside of it.